### PR TITLE
fix: Fix the go build and use CgoArrowAllocator to prevent incorrect garbage collection

### PR DESCRIPTION
**Pull request by kevjumba**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: docs/reference/feature-servers/go-feature-retrieval.md [hunk 1]
```diff
@@ -16,8 +16,19 @@ However, some additional dependencies are required for Go <-> Python interoperab
 ```
 pip install feast[go]
 ```
+You will also have to install the apache-arrow c++ libraries.
+For macos, run `brew install apache-arrow`.
+For linux users, you have to install `libarrow-dev`.
```
Comment (achals) [inline docs/reference/feature-servers/go-feature-retrieval.md:21]: Can you explain why this is the case?

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Comment by kevjumba** [comment]
fyi merging w/o integration test passing bc we need to update the integration test workflow.

**Pull request merged**